packages/terra-list/src/SingleSelectList.jsx

```diff
@@ -34,7 +39,13 @@ class SingleSelectList extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
-    this.state = { selectedIndex: SelectableList.Utils.initialSingleSelectedIndex(this.props.children) };
+    if (!this.props.hasSections) {
```

Comment:
Not sure about this if condition here, I believe this could lead to some issues if, for whatever reason, someone switched between `true` / `false` for the `hasSections` prop as this only sets state on initialization.

I'd suggest using [getderivedstatefromprops](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) instead. [More info on updating state based on props](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#updating-state-based-on-props).

Reply:
@bjankord Do you think, this should be implemented for SelectableList as well? Because, I think, terra lets consumers consume the SelectableList component and implement their own selection logic.